File: isogram.js
var isIsogram = (word) =>{
//an empty object to  store letters in the word and their count
var letterStore = {} 
//check if word is null and return true
var newWord = word.toLowerCase()
if(word.length === 0){
  console.log("This is an empty string")
}
var count = 0
//convert the input string to lower case
for(var i =0;i< newWord.length;i++){
  var letterInWord = newWord.charAt(i)
  if(letterStore[letterInWord]){
      letterStore[letterInWord] += letterStore[letterInWord];}
      else{
      letterStore[letterInWord] =1 
  }
}
if(letterStore[letterInWord] > 1){
  	console.log( newWord + " is not an isogram")
  }
else{
  console.log( newWord + " is an isogram")
}
}
//function test
isIsogram("asaaa")

File: isogram.py
def checkIsogram(word):
 
  count = 0
  newWord = word.lower()	
#check for an empty string
  if(len(newWord) == 0):
	print "this is an empty string"
  for i in newWord:
    if newWord.count(i) > 1:
      count = count + 1
  if ((count == 0) and (len(newWord) != 0)):
    print "word is isogram"
  if(count > 1):    
    print "word is not isogram"
	
word = raw_input("please enter a word >> :")
#function check
checkIsogram(word)
